proposal/eesss34690/README.rst

```diff
@@ -0,0 +1,87 @@
+Proposal for FFT Computing Optimization base on Columnar Array (CA-FFT)
+=============================================================================
+
+Basic information
+=================
+
+The GitHub repository is at: (URL)
+
+Problem to solve
+================
+
+Computing `Fast Fourier Transform <https://en.wikipedia.org/wiki/Fast_Fourier_transform>`__ is very important for digital recording, sampling. It is an algorithm computing the discrete Fourier transform (DFT) of a sequence. Since the sequences usually include lots of data and the computation of FFT is heavy, fasten up the solving speed is crucial for saving caching memory and also optimizing the computing speed.
+
+Currently the most common method to do FFT is ``Cooley-Tukey algorithm``. In this algorithm, it keeps breaking down a DFT of any size :math:`N = m \\times n` into smaller DFTs of sizes :math:`m` and :math:`n`, along with :math:`O(N)` multiplications by twiddle factor (typically either :math:`m` or :math:`n` is a small factor). Therefore, the DFT can be expressed as below:
+
+=============================================================================
+Proposal for FFT Computing Optimization base on Columnar Array (CA-FFT)
+=============================================================================
+
+Basic information
+=================
+
+The GitHub repository is at: (URL)
+
+Problem to solve
+================
+
+Computing `Fast Fourier Transform <https://en.wikipedia.org/wiki/Fast_Fourier_transform>`__ is very important for digital recording, sampling. It is an algorithm computing the discrete Fourier transform (DFT) of a sequence. Since the sequences usually include lots of data and the computation of FFT is heavy, fasten up the solving speed is crucial for saving caching memory and also optimizing the computing speed.
+
+Currently the most common method to do FFT is ``Cooley-Tukey algorithm``. In this algorithm, it keeps breaking down a DFT of any size :math:`N = m \times n` into smaller DFTs of sizes :math:`m` and :math:`n`, along with :math:`O(N)` multiplications by twiddle factor (typically either :math:`m` or :math:`n` is a small factor). Therefore, the DFT can be expressed as below:
+
+.. math::
+
+    DFT_{mn}= (DFT_m \otimes I_n)D_{m,n}(I_m \otimes DFT_n)L_m^{mn} \\
+    D_{m,n} = \oplus_{j=0}^{m-1}diag([w_{mn}^i]_{0\leq i \lt n})^j
+
+To optimize the computation, there are several categories that can implement. For example, we can optimize the cache usage for the complex computation. This can be done by ``blocking`` and ``buffering`` the data. With ``blocking``, we can operate the whole data with multiple same-size subset. Therefore, data loaded into the faster levels of the memory are reused. Because we are dealing with multi-dimensional data, the data which are closed to each other may be far away in memory. As the result, using ``buffering`` to copy the desired block into a columnar array and boost the performance. 
```

Comment:
We will practicing blocking (aka tiling) later in this course.  The implementation is (very) time-consuming.  Please do the following two things in your planning:

1. Rain check.  Stage your development so that if code for blocking cannot complete in time, you still have good results to show.  This is a must-do for real-world development.
2. More time.

Reply:
I will leave this part more time for developing. Also come up a backup plan which may still improve the performance.